src/serac/physics/solid_functional.hpp

```diff
@@ -249,12 +249,174 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
+            // If the normal is not already found, add it to the list
+            attribute_normals.emplace(att, normal);
+          } else {
+            // Otherwise check that it is consistent with previously found normals
+            auto& other_normal = found_attr->second;
+            for (int comp = 0; comp < normal.Size(); ++comp) {
+              SLIC_ERROR_IF(std::abs(std::abs(other_normal(comp)) - std::abs(normal(comp))) > 1.0e-7,
+                            "The normal vectors on an attribute surface are not constant in the sliding wall boundary "
+                            "condition.");
+            }
+          }
+        }
+      }
+    }
+
+    // Reduce the normals across MPI ranks to ensure consistency
+    for (int attr : slide_bdr) {
+      // Create structures to denote if a processor computed a normal and if so, the computed normal
+      auto         found_attr = attribute_normals.find(attr);
+      int8_t       found      = 0;
+      mfem::Vector normal(dim);
+      normal = 0.0;
+      if (found_attr != attribute_normals.end()) {
+        found  = 1;
+        normal = found_attr->second;
+      }
+
+      std::vector<int>    rank_contains_slide_wall(static_cast<std::size_t>(mpi_size_));
+      std::vector<double> rank_normals(static_cast<std::size_t>(dim * mpi_size_));
+
+      // Gather the computed normals and marker arrays across all processors
+      MPI_Allgather(&found, 1, MPI_INT8_T, rank_contains_slide_wall.data(), 1, MPI_CXX_BOOL, mesh_.GetComm());
+      MPI_Allgather(normal.GetData(), dim, MPI_DOUBLE, rank_normals.data(), dim, MPI_DOUBLE, mesh_.GetComm());
+
+      for (int i = 0; i < mpi_size_; ++i) {
+        // Check if each rank computed a normal
+        if (rank_contains_slide_wall[static_cast<std::size_t>(i)] == 1) {
+          if (found == 1) {
+            // If it did and we have a stored one, check that they are consistent.
+            for (int comp = 0; comp < normal.Size(); ++comp) {
+              SLIC_ERROR_IF(std::abs(std::abs(rank_normals[static_cast<std::size_t>(comp + (i * dim))]) -
+                                     std::abs(normal(comp))) > 1.0e-7,
+                            "The normal vectors on an attribute surface are not constant in the sliding wall boundary "
+                            "condition.");
+            }
+          } else {
+            // Otherwise, store the off-processor normal in the attribute normal container
+            found = 1;
+            for (int comp = 0; comp < normal.Size(); ++comp) {
+              normal(comp) = rank_normals[static_cast<std::size_t>(comp + (i * dim))];
+            }
+            attribute_normals.emplace(attr, normal);
+          }
+        }
+      }
+    }
+
+    // Loop over the computed attribute nomral vectors
+    for (auto& normal : attribute_normals) {
+      SLIC_ERROR_ROOT_IF(dim != 2 && dim != 3, "Dimension must be 2 or 3 for sliding wall boundary conditions.");
+
```

Comment:
The code above mentions that it ensures the normals are all the same, and then we loop over multiple normals here. Do we need a loop, or is there just one normal direction?

Reply:
The next step is adding multiple normals (i.e. multiple sliding walls).